src/main/java/com/kobylynskyi/graphql/codegen/mapper/TypeDefinitionToDataModelMapper.java

```diff
@@ -79,20 +69,27 @@ private static ParameterDefinition merge(ParameterDefinition typeDef, ParameterD
         ExtendedDocument document = mappingContext.getDocument();
 
         Map<String, Object> dataModel = new HashMap<>();
+        Collection<ParameterDefinition> fields = getFields(mappingContext, definition, document);
+        //kotlin not support empty
+        if (mappingContext.getGeneratedLanguage().equals(GeneratedLanguage.KOTLIN) &&
+                fields.isEmpty()) {
+            throw new UnableToCreateFileException(new Exception("Type requires at least one field to use in Kotlin."));
+        }
```

Comment:
Again, we just need to use a regular `class` or `object` in this case in Kotlin, just not a `data class`.